### DISCUSSION TITLE
White Balancing PDFs

### DISCUSSION THREAD

**Discussion by thejamesheard** [ImageMagick/ImageMagick#4849] [Help]
I'm a grad student and deal with bad scans every day... It would be really convenient to string together some scripts so that I can automatically white balance PDFs. I came across [Fred's ImageMagick autowhite script](http://www.fmwconcepts.com/imagemagick/autowhite/index.php), which seems to successfully do what I'm looking for. However... The output PDF is postage stamp sized. After some Googling I went through the script and added -density 300 after every instance of 'convert' but the output remains the same. I'm barely literate when it comes to scripting but it seems like with just a little more work I'm on my way to white balancing hundred page poorly scanned PDFs from the National Archives. I've attached the autowhite script, with my -density and -quality additions, as well as two test PDFs to show the results. Any help is greatly appreciated!

Thanks.

[autowhite.txt](https://github.com/ImageMagick/ImageMagick/files/8098611/autowhite.txt)
[TEST_IN.pdf](https://github.com/ImageMagick/ImageMagick/files/8098612/TEST_IN.pdf)
[TEST_OUT.pdf](https://github.com/ImageMagick/ImageMagick/files/8098613/TEST_OUT.pdf)

**Comment by fmw42** [comment]
Please contact me off-line (fmw at alink dot net). Please take a look at my my textcleaner script. That is likely what you need. I will have to add a density argument in it to handle PDF files.
